versioning/BRM/extentmap.cpp

```diff
@@ -782,68 +803,68 @@ int ExtentMap::setMaxMin(const LBID_t lbid,
+                   << " OID=" << emEntry.fileID << " min=" << min << " max=" << max
+                   << "seq=" << seqNum;
+                log(os.str(), logging::LOG_TYPE_DEBUG);
+            }
 
 #endif
 
-                if (curSequence == seqNum)
-                {
-                    makeUndoRecord(&fExtentMap[i], sizeof(struct EMEntry));
-                    fExtentMap[i].partition.cprange.hi_val = max;
-                    fExtentMap[i].partition.cprange.lo_val = min;
-                    fExtentMap[i].partition.cprange.isValid = CP_VALID;
-                    incSeqNum(fExtentMap[i].partition.cprange.sequenceNum);
-                    return 0;
-                }
-                //special val to indicate a reset--used by editem -c.
-                //Also used by COMMIT and ROLLBACK to invalidate CP.
-                else if (seqNum == -1)
-                {
-                    makeUndoRecord(&fExtentMap[i], sizeof(struct EMEntry));
-                    // We set hi_val and lo_val to correct values for signed or unsigned
-                    // during the markinvalid step, which sets the invalid variable to CP_UPDATING.
-                    // During this step (seqNum == -1), the min and max passed in are not reliable
-                    // and should not be used.
-                    fExtentMap[i].partition.cprange.isValid = CP_INVALID;
-                    incSeqNum(fExtentMap[i].partition.cprange.sequenceNum);
-                    return 0;
-                }
-                else
-                {
-                    return 0;
-                }
+            if (curSequence == seqNum)
+            {
+                makeUndoRecordRBTree(UndoRecordType::DEFAULT, emEntry);
+                emEntry.partition.cprange.hi_val = max;
+                emEntry.partition.cprange.lo_val = min;
+                emEntry.partition.cprange.isValid = CP_VALID;
+                incSeqNum(emEntry.partition.cprange.sequenceNum);
+                return 0;
+            }
+            // special val to indicate a reset--used by editem -c.
+            // Also used by COMMIT and ROLLBACK to invalidate CP.
+            else if (seqNum == -1)
+            {
+                makeUndoRecordRBTree(UndoRecordType::DEFAULT, emEntry);
+                // We set hi_val and lo_val to correct values for signed or unsigned
+                // during the markinvalid step, which sets the invalid variable to CP_UPDATING.
+                // During this step (seqNum == -1), the min and max passed in are not reliable
+                // and should not be used.
+                emEntry.partition.cprange.isValid = CP_INVALID;
+                incSeqNum(emEntry.partition.cprange.sequenceNum);
+                return 0;
+            }
+            else
+            {
+                return 0;
             }
         }
     }
 
-    if (emLocked)
-        releaseEMEntryTable(WRITE);
-
     if (emIndexLocked)
         releaseEMIndex(WRITE);
 
+    if (emLocked)
```

Comment:
You are changing the order here. IMHO this might open a tiny window for a deadlock.

Reply:
It seems like it was a bug, because we do lock in following order:
1. lockEmEntryTable
2. lockEmIndex

So the release should be
1. releaseEmIndex
2. releaseEmEntryTable

Reply:
Correct. Thx for fixing my mistake.